server/src/main/java/org/elasticsearch/action/admin/indices/alias/IndicesAliasesResponse.java

```diff
@@ -0,0 +1,136 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.indices.alias;
+
+import org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequest.AliasActions;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public class IndicesAliasesResponse extends AcknowledgedResponse {
+
+    public static final IndicesAliasesResponse NOT_ACKNOWLEDGED = new IndicesAliasesResponse(false, false, List.of());
+    public static final IndicesAliasesResponse ACKNOWLEDGED_NO_ERRORS = new IndicesAliasesResponse(true, false, List.of());
+
+    private static final String ACTION_RESULTS_FIELD = "action_results";
+    private static final String ERRORS_FIELD = "errors";
+
+    private final List<AliasActionResult> actionResults;
+    private final boolean errors;
+
+    protected IndicesAliasesResponse(StreamInput in) throws IOException {
+        super(in);
+        this.errors = in.readBoolean();
+        this.actionResults = in.readCollectionAsList(AliasActionResult::new);
+    }
+
+    private IndicesAliasesResponse(boolean acknowledged, boolean errors, final List<AliasActionResult> actionResults) {
+        super(acknowledged);
+        this.errors = errors;
+        this.actionResults = actionResults;
+    }
+
+    public static IndicesAliasesResponse build(final List<AliasActionResult> actionResults) {
+        final boolean errors = actionResults.stream().anyMatch(a -> a.success == false);
+        return new IndicesAliasesResponse(true, errors, actionResults);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeBoolean(errors);
+        out.writeCollection(actionResults);
+    }
+
+    @Override
+    protected void addCustomFields(XContentBuilder builder, Params params) throws IOException {
+        builder.field(ERRORS_FIELD, errors);
+        // if there are no errors, don't provide granular list of results
+        if (errors) {
+            builder.field(ACTION_RESULTS_FIELD, actionResults);
+        }
+    }
+
+    public static class AliasActionResult implements Writeable, ToXContentObject {
+        public static AliasActionResult REMOVE_MISSING = new AliasActionResult(
+            AliasActions.Type.REMOVE,
+            false,
+            "aliases_not_found_exception"
+        );
+
+        private final AliasActions.Type actionType;
```

Comment:
It might make sense to return the full action from the request rather than just the type. Thoughts?

Reply:
Yeah, ++, nobody's going to remember what order they submitted what in, so even though you're returning things in the same order, you need to include enough information in the response for them to understand what failed.

Reply:
Does **type** (add/remove), **indices**, and **aliases** seem like enough information? After some digging, I've found returning the original request is problematic because it gets modified. I can add other fields that are important, but it seems better to return fewer fields, as adding fields is non-breaking, but removing/update fields is a breaking change.